### PR TITLE
Predict probabilities

### PR DESCRIPTION
**Pull request by nbokulich**
All classifiers have a new output: predicted class probabilities. In other words, the predicted likelihood that each test sample belongs to each sample class. Useful for all sorts of things.

Specify a new format: `ProbabilitiesFormat`

Added tests for the format, and for replication of expected probabilities.

### CODE REVIEW

File: q2_sample_classifier/_format.py [hunk 1]
```diff
@@ -163,3 +163,15 @@ def _validate_(self, level):
 ImportanceDirectoryFormat = model.SingleFileDirectoryFormat(
     'ImportanceDirectoryFormat', 'importance.tsv',
     ImportanceFormat)
+
+
+class ProbabilitiesFormat(ImportanceFormat):
```
Comment (thermokarst) [inline q2_sample_classifier/_format.py]: How about instead we create a new base class and have both of these formats inherit from it?

File: q2_sample_classifier/_transformer.py [hunk 2]
```diff
@@ -65,13 +66,14 @@ def _4(data: pd.Series) -> (PredictionsFormat):
 def _5(ff: PredictionsFormat) -> (pd.Series):
     with ff.open() as fh:
         df = _read_dataframe(fh)
-        return df.iloc[:, 0]
+        return pd.to_numeric(df.iloc[:, 0], errors='ignore')
```
Comment (thermokarst) [inline q2_sample_classifier/_transformer.py]: What was going on that prompted this change? Do we need to add a test case?

File: q2_sample_classifier/classify.py [hunk 3]
```diff
@@ -262,20 +266,29 @@ def predict_base(table, sample_estimator, n_jobs):
...
-                           n_jobs: int = defaults['n_jobs']) -> pd.Series:
+                           n_jobs: int = defaults['n_jobs']) -> (
+                            pd.Series, pd.DataFrame):
     return predict_base(table, sample_estimator, n_jobs)
 
 
 def predict_regression(table: biom.Table, sample_estimator: Pipeline,
                        n_jobs: int = defaults['n_jobs']) -> pd.Series:
-    return predict_base(table, sample_estimator, n_jobs)
+    # we only return the predictions, not the probabilities, which are empty
```
Comment (thermokarst) [inline q2_sample_classifier/classify.py]: :cool:

File: q2_sample_classifier/_transformer.py (hunk 2 above)
Comment (thermokarst) [reply q2_sample_classifier/_transformer.py]: Oh is this the same changeset as https://github.com/qiime2/q2-sample-classifier/pull/158? If so, you can address my questions there.

Comment (nbokulich) [reply q2_sample_classifier/_transformer.py]: yes sorry need to rebase (forgot I had pulled in those changes here), will respond over there

File: q2_sample_classifier/_format.py (hunk 1 above)
Comment (nbokulich) [reply q2_sample_classifier/_format.py]: 👍

**Comment by nbokulich** [comment]
thanks @thermokarst ! I added the base class `MultiColumnNumericFormat` per your request.

I rebased in an attempt to clean up the history but looks like that failed...

File: q2_sample_classifier/_format.py [hunk 4]
```diff
@@ -120,7 +120,7 @@ def _validate_(self, level):
     PredictionsFormat)
 
 
-class ImportanceFormat(model.TextFileFormat):
+class MultiColumnNumericFormat(model.TextFileFormat):
```
Comment (thermokarst) [inline q2_sample_classifier/_format.py]:
```suggestion
class _MultiColumnNumericFormat(model.TextFileFormat):
```

Let's mark this as "private", that way an endeavoring developer doesn't get too carried away with this format.

Comment (nbokulich) [reply q2_sample_classifier/_format.py]: 👍

Comment (nbokulich) [reply q2_sample_classifier/_format.py]: thanks for breaking my tests, dude 😆

**Pull request merged**